extensions/hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3Repository.java

```diff
@@ -220,4 +235,25 @@ private static DigestOutputStream openFileOutputStream(final File file, final Me
     private boolean exists(final String sha1) {
         return amazonS3.doesObjectExist(s3Properties.getBucketName(), sha1);
     }
+
+    @Override
+    public void deleteByTenant(final String tenant) {
+        final String folder = sanitizeTenant(tenant);
+
+        LOG.info("Deleting S3 object folder (tenant) from bucket {} and key {}", s3Properties.getBucketName(), folder);
+
+        // Delete artifacts
+        ObjectListing objects = amazonS3.listObjects(s3Properties.getBucketName(), folder + "/");
+        do {
+            for (final S3ObjectSummary objectSummary : objects.getObjectSummaries()) {
+                amazonS3.deleteObject(s3Properties.getBucketName(), objectSummary.getKey());
+            }
+            objects = amazonS3.listNextBatchOfObjects(objects);
```

Comment:
Don't you have to call `objects.isTruncated()` first before requesting the next bunch of objects?